app/budget/services/actions/budgetActions.js

```diff
@@ -427,7 +427,92 @@ class BudgetActions {
+				// Append proper year to term
+				newExpenseItem.termCode = newExpenseItem.termCode ? TermService.termToTermCode(newExpenseItem.termCode, year) : null;
 
+				BudgetService.createExpenseItem(newExpenseItem, budgetScenarioId).then(function (newExpenseItem) {
+					window.ipa_analyze_event('budget', 'expense item');
+
+					var action = {
+						type: ActionTypes.CREATE_EXPENSE_ITEM,
+						payload: newExpenseItem
+					};
+					$rootScope.$emit('toast', { message: message || "Created expense", type: "SUCCESS" });
+					BudgetReducers.reduce(action);
+
+					// Close modal
+					self.closeAddExpenseItemModal();
+					//BudgetCalculations.calculateExpenseItems(); TODO support if needed
+					BudgetCalculations.calculateTotalCost();
+				}, function () {
+					$rootScope.$emit('toast', { message: "Could not create expense.", type: "ERROR" });
+				});
+			},
+			updateExpenseItem: function (expenseItem) {
+				var self = this;
+
+				// Create instead of update if appropriate
+				if (expenseItem.id == null || expenseItem.id == 0) {
+					this.createExpenseItem(expenseItem, expenseItem.budgetScenarioId);
+					return;
+				}
+
+				var year = BudgetReducers._state.ui.year;
+				// Ensure amount is properly formatted as a float
+				expenseItem.amount = expenseItem.amount ? parseFloat(expenseItem.amount) : null;
+				// Append proper year to term if needed
+				if (expenseItem.termCode && expenseItem.termCode.length < 6) {
+					expenseItem.termCode = TermService.termToTermCode(expenseItem.termCode, year);
+				}
+
+
+				BudgetService.updateExpenseItem(expenseItem, expenseItem.budgetScenarioId).then(function (results) {
+					window.ipa_analyze_event('budget', 'expense item');
+
+					var action = {
+						type: ActionTypes.UPDATE_EXPENSE_ITEM,
+						payload: results
+					};
+					$rootScope.$emit('toast', { message: "Saved expense", type: "SUCCESS" });
+					BudgetReducers.reduce(action);
+
+					// Close modal
+					self.closeAddExpenseItemModal();
+					BudgetCalculations.calculateTotalCost();
+				}, function () {
+					$rootScope.$emit('toast', { message: "Could not save expense.", type: "ERROR" });
+				});
+			},
+			deleteExpenseItem: function(expenseItem) {
+				// If the expenseItem is based on a teachingAssignment, do not delete it, instead mark it as hidden
+				if (expenseItem.teachingAssignmentId > 0) {
+					expenseItem.hidden = true;
+					this.updateExpenseItem(expenseItem);
+					return;
+				}
+
+				BudgetService.deleteExpenseItem(expenseItem).then(function (expenseItemId) {
+					window.ipa_analyze_event('budget', 'expense deleted');
+
+					var action = {
+						type: ActionTypes.DELETE_EXPENSE_ITEM,
+						payload: {
+							expenseItemId: expenseItemId
+						}
+					};
+
+					$rootScope.$emit('toast', { message: "Deleted expense", type: "SUCCESS" });
+					BudgetReducers.reduce(action);
+					//BudgetCalculations.calculateExpenseItems(); TODO support if needed
```

Comment:
It looks like deleting item(s) doesn't update the values in the navbar or the Summary table. Is that something we can implement?

I noticed it doesn't do it for the Funds tab and was considering updating that behavior.

Reply:
I updated it for the other costs.  Let me know if we want to update the funds on this branch as well.

Reply:
Mmm, let's keep it separate. I already made a Jira issue for it.